File: 1046-last-stone-weight/1046-last-stone-weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        sort(stones.begin(),stones.end());
        while(stones.size() > 1){
            int a = stones[stones.size()-1];
            int b = stones[stones.size()-2];
            stones.pop_back();
            stones.pop_back();
            stones.push_back(a-b);
            sort(stones.begin(),stones.end());
        }
        return stones[0];
        // return 0;   
    }
};

File: 1046-last-stone-weight/1046-last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        heap=[-i for i in stones]
        heapify(heap)
        
        while len(heap)>1:
            # print(heap)
            a,b=heappop(heap),heappop(heap)
            if a==b: continue
            elif a<b:heappush(heap,a-b)
            else: heappush(heap,b-a)
                
        if heap:  return -heap[0]
        return  0
            